appcontext/context_test.go

```diff
@@ -0,0 +1,60 @@
+package appcontext
+
+import (
+	"testing"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/pkg/startup"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/coredata"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestComplete(t *testing.T) {
+	ctx := Context{}
+	testData := "output data"
+	ctx.Complete([]byte(testData))
+	assert.Equal(t, []byte(testData), ctx.OutputData)
+}
+
+var eventClient coredata.EventClient
+
+func init() {
+	params := types.EndpointParams{
+		ServiceKey:  clients.CoreDataServiceKey,
+		Path:        clients.ApiEventRoute,
+		UseRegistry: false,
+		Url:         "http://test" + clients.ApiEventRoute,
+		Interval:    1000,
+	}
+	eventClient = coredata.NewEventClient(params, startup.Endpoint{RegistryClient: nil})
+
+}
+func TestMarkAsPushedNoEventIdOrChecksum(t *testing.T) {
+	ctx := Context{}
+	err := ctx.MarkAsPushed()
+
+	assert.Equal(t, "No EventID or EventChecksum Provided", err.Error())
+}
+
+func TestMarkAsPushedNoChecksum(t *testing.T) {
+	ctx := Context{
+		EventChecksum: "checksumValue",
+		CorrelationID: "correlationId",
+		EventClient:   eventClient,
+	}
+	err := ctx.MarkAsPushed()
+
+	assert.Contains(t, err.Error(), "http://test/api/v1/event/checksum/checksumValue")
```

Comment:
Same err == nil issue, Trevor mentioned there are some mocks for core data somewhere that could be used here?

Reply:
Do you know where/what they are? All i cared about was the correct URL being called, which err will always be not nil when these tests are run in the pipeline as the URL it hits wont be available

Reply:
Nope I don't. Ask Trevor.